### PR TITLE
Reset random number generator before evaluating constants per worker.

### PR DESCRIPTION
**Pull request by lemmy**
Fixes Github issue #866
https://github.com/tlaplus/tlaplus/issues/866

[Bug][TLC]

### CODE REVIEW

**Comment by lemmy** [comment]
This lacks a test case that reproduces the bug.

**Comment by FedericoPonzi** [comment]
Sample repro:

This is the spec and model:
```
---------------------------- MODULE Github866 ---------------------------
EXTENDS Naturals, TLC, FiniteSets
VARIABLES x
vars == << x >>

X == {RandomElement({1,2,3})}

Init == x = 0
Lbl_1 == /\ x < 100 \* Generate some states to ensure both workers have work to do.
         /\ TLCSet(42, X)
         /\ x' = x + 1

Next == Lbl_1

Spec == Init /\ [][Next]_vars

==========================================================================
---------------------------- CONFIG Github866 ----------------------------
SPECIFICATION Spec
=============================
```

This is the test:
```
package tlc2.tool;

import org.junit.Test;
import tlc2.TLCGlobals;
import tlc2.output.EC;
import tlc2.tool.liveness.ModelCheckerTestCase;
import tlc2.value.IValue;

import java.io.IOException;
import java.util.List;
import java.util.HashSet;

import static org.junit.Assert.assertTrue;
import static org.junit.Assert.assertFalse;
import static org.junit.Assert.assertEquals;

public class Github866Test extends ModelCheckerTestCase {

	public Github866Test() {
		super("Github866", new String[] { "-config", "Github866.tla" }, EC.ExitStatus.SUCCESS);
	}
    @Override
    protected int getNumberOfThreads(){
        return 2;
    }

	@Test
	public void testSpec() throws IOException {
        assertTrue(recorder.recorded(EC.TLC_FINISHED));
        assertFalse(recorder.recorded(EC.GENERAL));
        final List<IValue> allValue = TLCGlobals.mainChecker.getAllValue(42);
        assertEquals(1, new HashSet(allValue).size());
	}
}
```
The test was failing before your fix, and passed after. I'm generating some states just to ensure there is enough work for both threads, in some runs a single thread was able to take all the work and finish (resulting with a "null" for the other thread). I know it's not ideal, but hopefully it won't be flaky.

**Comment by lemmy** [comment]
For whatever reason, Github won't let me request a review from @FedericoPonzi.

File: tlatools/org.lamport.tlatools/test-model/Github866.tla [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+VARIABLES x
+
+Init ==
+  x = 0
+
+Next ==
+  \* Keep going until all workers have evaluated next.
+  /\ \E w \in 1..TLCGet("config").worker: 
+          TLCGet("all")[r][w] = {0}
```
Comment (FedericoPonzi) [inline tlatools/org.lamport.tlatools/test-model/Github866.tla:32]: This is very nice!

**Review by FedericoPonzi — APPROVED** [review]
LGTM.

**Review by Calvin-L — APPROVED** [review]

**Pull request merged**